torchaudio/sox_effects.py

```diff
@@ -84,6 +92,9 @@ def __init__(self, normalization=True, channels_first=True, out_siginfo=None, ou
         self.normalization = normalization
         self.channels_first = channels_first
 
+        # Define in __init__ to avoid calling at import time
+        self.EFFECTS_AVAILABLE = set(effect_names())
```

Comment:
This is BC-breaking in the sense that users could check the `EFFECTS_AVAILABLE` without creating an instance.

Reply:
Why is this change necessary?

Reply:
If `effect_names` is a class variable, the python function `effect_names` gets run at import time, which then imports `_torch_sox` for `_torch_sox.get_effect_names`.

Reply:
So you're concerned that it's possible that someone imports torchaudio, checks this variable and finds there are no effects available? But then once that persons uses load or such once it'll be populated?

Reply:
(Corrected typo in last response.)

I was just highlighting that it won't be able to do `SoxEffectsChain.EFFECTS_AVAILABLE` anymore. Instead, a user will have to do `SoxEffectsChain().EFFECTS_AVAILABLE`. I do not think anyone was doing the first. 

Calling load won't change this behavior.

Reply:
Ok that seems fine, I'd only mention it in the release notes for completeness.

Reply:
Added to PR description. Resolving discussion.